server/src/main/java/org/elasticsearch/cluster/routing/allocation/DiskThresholdSettings.java

```diff
@@ -55,6 +83,20 @@ public class DiskThresholdSettings {
         Setting.Property.Dynamic,
         Setting.Property.NodeScope
     );
+    public static final Setting<ByteSizeValue> CLUSTER_ROUTING_ALLOCATION_DISK_FLOOD_STAGE_MAX_HEADROOM_SETTING = new Setting<>(
+        "cluster.routing.allocation.disk.watermark.flood_stage.max_headroom",
+        (settings) -> {
+            if (CLUSTER_ROUTING_ALLOCATION_DISK_FLOOD_STAGE_WATERMARK_SETTING.exists(settings)) {
+                return "-1";
+            } else {
+                return "20GB";
```

Comment:
I think this should default to `50GB`. That's a pretty typical target size for a shard, and some operations like a force-merge may temporarily double the size of a shard even if writes are blocked, so anything less than `50GB` here means there's a much bigger risk of running out of space.

When we address #88606 I expect we can relax this down to `20GB` or less.

Reply:
Nice, sounds reasonable! I can change it to 50GB. Note that the frozen tier flood stage watermark setting will still have a default value of 20GB. Is that OK? Or would we expect similar merging issues for the frozen tier?

@henningandersen please comment if you have a different suggestion than changing the default flood max headroom to 50GB. For the moment, I will go ahead and make a commit for this change.

> When we address https://github.com/elastic/elasticsearch/issues/88606 I expect we can relax this down to 20GB or less.

Should we create a new issue for this or maybe you could add a comment in that issue to remember if this PR gets merged to see if it makes sense to decrease the max headroom?

Reply:
It will be ok to run closer to the limit in the frozen tier because the indices in this tier are completely immutable so they'll never do merges.

> Should we create a new issue for this

Good point. I edited the issue description to keep hold of this observation.

Reply:
Sounds good. I changed the default from 20GB to 50GB for the flood max headroom. Please feel free to review and unresolve the conversation if there is more feedback. Thanks!

Reply:
Unresolved this to ensure we revisit the topic of whether the flood stage max headroom should be bigger than 50GB.

Reply:
I discussed this offline with David and we think we should raise the flood stage max headroom to 100GB and shift the other watermarks's maxheadroom 50GB as well (ie., high=150, low=200 GB).

This ensures that we will not run out of disk space for just 1-2 force merge to one segment.

Reply:
Hi @henningandersen , I applied your suggested new default values. Please feel free to review and unresolve this conversation if you feel it's not handled sufficiently.